src/app/courses/pages/courses/courses.component.ts

```diff
@@ -54,28 +53,78 @@ export class CoursesComponent implements OnInit {
     this.loadMore();
   }
 
+  canAddCourse(): Boolean {
```

Comment:
The same as above about a lot of recalculations. You may set this simple logic into the template to check is user authenticated only (without any functions call).
This comment refers also to canAddCourse, canEditCourse, canDeleteCourse